build/build-viewer.js

```diff
@@ -140,12 +140,12 @@ async function compileJs() {
   const options = {
     output: {preamble: license}, // Insert license at top.
   };
-  const uglified = uglifyEs.minify(contents, options);
+  const uglified = Terser.minify(contents, options);
   if (uglified.error) {
     throw uglified.error;
   }
 
-  await safeWriteFileAsync(`${distDir}/src/viewer.js`, uglified.code);
+  await safeWriteFileAsync(`${distDir}/src/viewer.js`, /** @type {string} */ (uglified.code));
```

Comment:
maybe check `code` is defined (if not, throw error) instead of coerce?

Reply:
we already throw if uglified.error is true. When it's false code should exists.

Reply:
FWIW ward we have generally moved to a more catering-to-typescript style where impossible things that can never happen are still asserted in the code to make ts life easier and the assumption more explicit :)

Reply:
> we already throw if uglified.error is true. When it's false code should exists.

this is true but I agree with @connorjclark that casts make me nervous :)

What about making the check above this into
```js
if (uglified.error || !uglified.code) {
  throw uglified.error;
}
```
then it won't need a cast (the downside is it'll throw `undefined` if `error` and `code` are undefined, but in theory that should never happen and the error will still be handled :)

Reply:
> FWIW ward we have generally moved to a more catering-to-typescript style where impossible things that can never happen are still asserted in the code to make ts life easier and the assumption more explicit :)

Make sense! Thanks for clarifying. It has been a while since i've contributed so forgive me 😉